src/setfit/integrations.py

```diff
@@ -1,28 +1,51 @@
 import importlib.util
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Optional
 
 from .utils import BestRun
+from .trainer import SetFitTrainer
 
 
-if TYPE_CHECKING:
-    from .trainer import SetFitTrainer
+def is_optuna_available() -> bool:
+    """Check if Optuna is available."""
+    return importlib.util.find_spec("optuna") is not None
 
 
-def is_optuna_available():
-    return importlib.util.find_spec("optuna") is not None
+def default_hp_search_backend() -> Optional[str]:
+    """Get the default backend for hyperparameter search."""
+    return "optuna" if is_optuna_available() else None
+
+
+def run_hp_search_optuna(
+    trainer: SetFitTrainer, 
+    n_trials: int, 
+    direction: str, 
+    timeout: Optional[int] = None,
+    n_jobs: int = 1,
+    **kwargs,
+) -> BestRun:
+    """Run hyperparameter search with Optuna.
 
+    Args:
+        trainer: SetFitTrainer object to train and evaluate the model.
+        n_trials: Number of trials to run for hyperparameter search.
+        direction: Optimization direction for the study.
+        timeout: Maximum duration in seconds for running the trials.
+        n_jobs: Number of parallel jobs to run.
```

Comment:
```suggestion
        trainer (`SetFitTrainer`): SetFitTrainer object to train and evaluate the model.
        n_trials (`int`): Number of trials to run for hyperparameter search.
        direction (`str`): Optimization direction for the study, either "minimize" (default) or "maximize".
        timeout (`int`, *optional*): Maximum duration in seconds for running the trials.
        n_jobs (`int`, defaults to `1`): Number of parallel jobs to run.
```
Some nitpicks regarding the docstring, to match the style of the rest of the repo. This should help out when we finalize our documentation.